Calculator.py

```diff
@@ -17,6 +17,9 @@ def run_operation(number_1,  number_2, operation):
     elif operation == '/':
         print('{} / {} = '.format(number_1, number_2), end = '')
         return number_1 / number_2
+    elif operation == '%':
+        print('{} % {} = '.format(number_1, number_2), end = '')
+        return number_1 % number_2
     else:
         raise Exception('This operator is invalid. Try running the program again.')
 
```

Comment:
This is good error checking for the operation. Maybe you can extend this so it can check if the user enter numbers, and not letters or symbols when prompted from lines 3-4. Or you could change it so the user doesn't have the run the program again but they're prompted again instead.

Reply:
I was thinking the exact same thing. These lines are a good start, but it would be better for the user to not have to rerun the program each time an error arises

Reply:
I was thinking something similar, you could prompt the user for numbers again if you catch an exception.

Reply:
> I was thinking something similar, you could prompt the user for numbers again if you catch an exception.

Yes, I think it would be appropriate to allow the user an immediate chance to enter a new number as soon as an exception is caught for the best UX